src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -525,27 +524,31 @@ YlmSpherepack::first_and_second_derivative(
   return result;
 }
 
-YlmSpherepack::InterpolationInfoPerPoint::InterpolationInfoPerPoint(
+template <typename T>
+YlmSpherepack::InterpolationInfo<T>::InterpolationInfo(
     const size_t m_max, const std::vector<double>& pmm,
-    const std::array<double, 2>& target)
+    const std::array<T, 2>& target)
     : cos_theta(cos(target[0])),
       cos_m_phi(m_max + 1),
       sin_m_phi(m_max + 1),
-      pbar_factor(m_max + 1) {
-  const double sin_theta = sin(target[0]);
-  const double phi = target[1];
+      pbar_factor(m_max + 1),
+      num_points(get_size(target[0])) {
+  // do not use auto here, the blaze expression templates will create a dangling
+  // reference that is not caught
+  const T sin_theta = sin(target[0]);
+  const T phi = target[1];
```

Comment:
Doesn't this line just copy the `DataVector`? I think just make it `const auto& phi`